gdal/configure.ac

```diff
@@ -2153,6 +2153,65 @@ AC_SUBST(SOSI_ENABLED, $SOSI_ENABLED)
 AC_SUBST(SOSI_LIB, $SOSI_LIB)
 AC_SUBST(SOSI_INC, $SOSI_INC)
 
+dnl ---------------------------------------------------------------------------
+dnl JNI
+dnl ---------------------------------------------------------------------------
+
+AC_ARG_WITH(jni,[  --with-jni[=ARG]        Include JVM JNI support (ARG=Path or no)],,)
+
+JNI_ENABLED=no
+
+if test -z "$with_jni" -o "$with_jni" = "no"; then
+     AC_MSG_RESULT([JNI inclusion not requested.])
+else
+     AC_MSG_CHECKING([for libjvm.so in $with_jni])
+     AC_CHECK_LIB(jvm,JNI_GetCreatedJavaVMs,HAVE_JNI=yes,HAVE_JNI=no,-L$with_jni)
+     if test "$HAVE_JNI" = "yes"; then
+        AC_MSG_RESULT([JNI included.])
+        JNI_ENABLED=yes
+        JNI_LIB="-L$with_jni -ljvm"
+     else
+        AC_MSG_ERROR([Unable to include JNI.])
+     fi
+fi
+
+AC_SUBST(JNI_ENABLED,$JNI_ENABLED)
+AC_SUBST(JNI_LIB,$JNI_LIB)
+
+
+dnl ---------------------------------------------------------------------------
+dnl HDFS support
+dnl ---------------------------------------------------------------------------
+
+AC_ARG_WITH(hdfs,[  --with-hdfs[=ARG]        Include HDFS support (ARG=Path or no)],,)
+
+HDFS_ENABLED=no
+
+if test -z "$with_hdfs" -o "$with_hdfs" = "no" ; then
+     AC_MSG_RESULT([HDFS support not requested.])
+else
+     if test JNI_ENABLED="yes"; then
+          AC_MSG_CHECKING([for HDFS in $with_hdfs])
+          AC_CHECK_LIB(hdfs,hdfsConnect,HAVE_HDFS_LIB=yes,HAVE_HDFS_LIB=no,$JNI_LIB -L$with_hdfs/lib/native)
```

Comment:
so libhdfs is a C-api library that calls Java code ?

Reply:
Yes, that is correct.